### PR TITLE
tests: Added arm github runner to build snapd

### PR DESCRIPTION
**Pull request by maykathm**
https://warthogs.atlassian.net/browse/SNAPDENG-31561

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -853,21 +866,30 @@ jobs:
           # Add start line to filtered log
           echo "GRAFANA START: pr ${CHANGE_ID} attempt ${{ github.run_attempt }} run ${{ github.run_id }} group ${{ matrix.group }}" > "$FILTERED_LOG_FILE"
 
-    - name: Download built snap
+    - name: Download built snap (amd)
+      if: "!endsWith(matrix.group, 'arm') && !endsWith(matrix.group, '-fips')"
```
Comment (zyga) [inline .github/workflows/test.yaml:870]: CC @bboozzoo on FIPS

**Review by zyga — COMMENTED** [review]
I think this looks good based on my quick look at the diff. The only question I would ask now is about FIPS support. Why is that x86 specific?

**Comment by maykathm** [comment]
> I think this looks good based on my quick look at the diff. The only question I would ask now is about FIPS support. Why is that x86 specific?

I think that's more a question for @bboozzoo. Is it just that we don't yet have a spread image prepared with ARM+FIPS or is there another reason?

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -11,9 +11,12 @@ concurrency:
 
 jobs:
   snap-builds:
-    runs-on: ubuntu-22.04
+    runs-on: ${{ matrix.runner }}
     strategy:
       matrix:
+        runner:
```
[the remark this answers is not in the document]
Comment (maykathm) [reply .github/workflows/test.yaml:17]: Added. Thanks!

**Comment by bboozzoo** [comment]
I think it's fine to skip ARM+FIPS until we have a clear ask to add it. One less point of failure.

**Review by bboozzoo — APPROVED** [review]
Thanks

**Comment by bboozzoo** [comment]
The branch most likely needs a rebase. We'll also need to sync with @ernestl and @pedronis as the list of required checks will likely need to be updated

**Comment by maykathm** [comment]
This PR should land after the https://github.com/canonical/snapd/pull/14689 workflows reorganization PR.

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -13,11 +13,14 @@ jobs:
...
     with:
-      runs-on: ubuntu-22.04
+      runs-on: ${{ matrix.runs-on }}
       toolchain: ${{ matrix.toolchain }}
       variant: ${{ matrix.variant }}
     strategy:
       matrix:
+        runs-on:
+          - '["ubuntu-22.04"]'
+          - '["self-hosted", "Linux", "jammy", "ARM64", "large"]'
```
[the remark this answers is not in the document]
Comment (maykathm) [reply .github/workflows/test.yaml:23]: Sure. The list comes from IS; they're tags that they tagged runners with and we can use to identify what runners we want to use. So the capitalization choices and terms are theirs. Would something like "tags to identify self-hosted runners" work?

[the remark this answers is not in the document]
Comment (zyga) [reply .github/workflows/test.yaml:23]: If the list comes from IS just keep that as the comment.

File: .github/workflows/spread-tests.yaml [hunk 4]
```diff
@@ -133,21 +133,30 @@ jobs:
           # Add start line to filtered log
           echo "GRAFANA START: pr ${CHANGE_ID} attempt ${{ github.run_attempt }} run ${{ github.run_id }} group ${{ inputs.group }}" > "$FILTERED_LOG_FILE"
 
-    - name: Download built snap
+    - name: Download built snap (amd64)
       uses: actions/download-artifact@v4
-      if: "!endsWith(inputs.group, '-fips')"
+      if: "!contains(inputs.group, '-arm') && !endsWith(inputs.group, '-fips')"
```
Comment (zyga) [inline .github/workflows/spread-tests.yaml:138]: Should this say arm64 explicitly?
```suggestion
      if: "!contains(inputs.group, '-arm64') && !endsWith(inputs.group, '-fips')"
```

File: .github/workflows/spread-tests.yaml [hunk 5]
```diff
@@ -133,21 +133,30 @@ jobs:
...
       with:
-        name: snap-files-default-test
+        name: snap-files-amd64-default-test
         # eg. snapd_1337.2.65.1+git97.gd35b459_amd64.snap
         pattern: snapd_1337.*.snap
         path: "${{ github.workspace }}/built-snap"
 
-    - name: Download built FIPS snap
+    - name: Download built snap (arm64)
+      if: "contains(inputs.group, '-arm') && !endsWith(inputs.group, '-fips')"
```
Comment (zyga) [inline .github/workflows/spread-tests.yaml:146]: Ditto?

```suggestion
      if: "contains(inputs.group, '-arm64') && !endsWith(inputs.group, '-fips')"
```

File: .github/workflows/spread-tests.yaml [hunk 6]
```diff
@@ -133,21 +133,30 @@ jobs:
...
+        name: snap-files-arm64-default-test
+        pattern: snapd_1337.*.snap
+        # eg. snapd_1337.2.65.1+git97.gd35b459_amd64.snap
+        path: "${{ github.workspace }}/built-snap"
+
+    - name: Download built FIPS snap (amd64)
       uses: actions/download-artifact@v4
       # eg. ubuntu-fips
-      if: "endsWith(inputs.group, '-fips')"
+      if: "!contains(inputs.group, '-arm') && endsWith(inputs.group, '-fips')"
```
Comment (zyga) [inline .github/workflows/spread-tests.yaml:157]: Same

**Review by zyga — APPROVED** [review]
LGTM but it seems something else at another level also uses the "arm" name (or perhaps we just look for arm out of the arm64 suffix)

**Comment by maykathm** [comment]
## Failures

google-distro-2:opensuse-tumbleweed-64:tests/main/microk8s-smoke:edge - unrelated problem
FIPS - known issue

**Comment by ernestl** [comment]
Note: require snap-builds needs to be changed after this ticket is merged.

**Pull request merged**